### PR TITLE
feat: enables podman machine for linux

### LINKED ISSUE #5823: Support enabling Podman Machine detection on Linux
### Is your feature request related to a problem? Please describe

See: https://github.com/containers/podman-desktop/discussions/5762

There are cases where creating a podman machine on Linux would be beneficial, such as monitoring / viewing a "root" podman machine to be able to run root containers.

### Describe the solution you'd like

Enabling podman machine for linux.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: enables podman machine for linux

### What does this PR do?

* Enables Podman Machine detection for Linux
* Allows the ability for PD to detect and deploy to Podman machine
* Does not allow creation (you can only do this via CLI).

### Screenshot / video of UI

Screenshot from 2024-02-07 16-43-50

### What issues does this PR fix or reference?

Closes #5823

### How to test this PR?

1. `podman machine init` or `podman machine init --rootful` on the CLI
2. Open up PD
3. See that the machine is detected on the Providers list
4. Pull an image / Build an image and select "Podman Machine", it'll be
   successful.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
Honestly, only a few amount of changes needed to get this working correctly..

I'm assuming let's add a e2e test for this functionality? What are your thoughts @odockal

**Comment by afbjorklund** [comment]
This would probably look nicer, if it used the same units as `podman machine` does?

```
NAME                    VM TYPE     CREATED      LAST UP            CPUS        MEMORY      DISK SIZE
podman-machine-default  qemu        10 days ago  Currently running  4           2GiB        100GiB
```

**Comment by cdrage** [comment]
> This would probably look nicer, if it used the same units as `podman machine` does?
> 
> ```
> NAME                    VM TYPE     CREATED      LAST UP            CPUS        MEMORY      DISK SIZE
> podman-machine-default  qemu        10 days ago  Currently running  4           2GiB        100GiB
> ```

Might be a different issue? I believe we've had an issue opened for a while regarding the same units showing on the UI vs podman cli.

**Comment by afbjorklund** [comment]
I think the units are not an issue in Mac and Win, since you create them in the interface using the same units (GB).

So Podman Desktop creates a machine with 3.725GiB, and then it is shown in the GUI as 4.000GB

But it becomes an issue, when the machine is created with the CLI which uses the binary units (GiB).

* #3047

**Comment by cdrage** [comment]
Ready for preliminary review 👍 

e2e tests will be added after #5197 is merged in (and implemented for Linux too).

**Comment by odockal** [comment]
@cdrage Thanks for bringing this up. This totally deserves e2e test case, and yes, we can make use of the existing that are being worked on atm. It will need special infrastructure to run this tests on, if GHA ubuntu runner will not be enough. That needs to be tried out. Details in here: #6000

**Review by benoitf — COMMENTED** [review]
it looks like we're missing unit tests

I see you edited the comment of the test but I think it's missing stuff somewhere else no ?

Also the edited test has probably a wrong copy/paste
```ts
// Mock platform to be darwin
  vi.mocked(isMac).mockReturnValue(false);
```
as it's mostly saying it's not macOS

**Comment by cdrage** [comment]
> it looks like we're missing unit tests
> 
> I see you edited the comment of the test but I think it's missing stuff somewhere else no ?
> 
> Also the edited test has probably a wrong copy/paste
> 
> ```ts
> // Mock platform to be darwin
>   vi.mocked(isMac).mockReturnValue(false);
> ```
> 
> as it's mostly saying it's not macOS

Looks like I was in the process of adding the other test and didn't push it! I've added the other simple test.

e2e tests will be covered by #1475 as we do not have any tests right now for activate(), as we don't have any tests any other platform, so just edit capabilities has been added for unit tests.

File: extensions/podman/src/extension.spec.ts [hunk 1]
```diff
@@ -1060,6 +1059,24 @@ test('provider is registered without edit capabilities on Windows', async () =>
...
+  });
+  let registeredConnection: ContainerProviderConnection;
+  vi.mocked(provider.registerContainerProviderConnection).mockImplementation(connection => {
+    registeredConnection = connection;
+    return Disposable.from({ dispose: () => {} });
+  });
+  await extension.registerProviderFor(provider, machineInfo, undefined);
+  expect(registeredConnection).toBeDefined();
+  expect(registeredConnection.lifecycle).toBeDefined();
+  expect(registeredConnection.lifecycle.edit).toBeUndefined();
```
Comment (dgolovin) [inline extensions/podman/src/extension.spec.ts:1077]:
```suggestion
  expect(registeredConnection?.lifecycle?.edit).toBeUndefined();
```

**Review by dgolovin — APPROVED** [review]
LGTM with an nit (have not tested it)

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
@cdrage could you rebase so it passes the sonar check ? and merge

**Comment by afbjorklund** [comment]
@cdrage : Podman Desktop now goes into an endless "setup" loop, when the podman machine is not running

Will open a bug report about it.

EDIT:

* https://github.com/containers/podman-desktop/issues/6151

**Pull request merged**